acceptance/tests/lookup/lookup_rich_values.rb

```diff
@@ -31,44 +33,54 @@ def rich_data_test(options, context)
   end
 end
     FUNC
-    create_sitepp(master, tmp_environment, <<-SITE)
-      notify { "${unwrap(lookup('environment_key'))}": }
-    SITE
+  end
+
+  step "create puppet language lookup function in #{tmp_environment} module" do
+    on(master, "mkdir -p #{fq_tmp_environmentpath}/modules/some_mod/functions")
+    create_remote_file(master, "#{fq_tmp_environmentpath}/modules/some_mod/functions/rich_data_test2.pp", <<-FUNC)
+function some_mod::rich_data_test2($options, $context) {
+  {
+    "environment_key2" => Sensitive('#{sensitive_value_pp}'),
```

Comment:
This isn't supposed to work. All keys in a module must be prefixed with the module name. The reason it does work is that it is part of an environment hierarchy.
@hlindberg, what's your opinion on this. Should we allow the hierarchy in a hiera.yaml that resides in an environment to reference functions provided by a module?